### DISCUSSION TITLE
How to uninstall and delete everything?

### DISCUSSION THREAD

**Discussion by Siggy101** [teslamate-org/teslamate#1125] [General]
Does anyone have instructions on how to cleanly 'uninstall' Teslamate from a Docker setup?
I have decided not to use TeslaMate anymore and would like to remove all containers and would like to be sure I am tidying up all orphaned data from my server.  I cannot see any removal instructions on the docs site.
Is it enough to just 'remove' the containers in Portainer or will I have orphaned volumes that will need manually removing?  For info, I followed the official docker-compose install route: (https://docs.teslamate.org/docs/installation/docker/).

**Comment by Adeian** [comment]
After losing a years worth of data after an update and getting no help from here I feel you.  :)

You will have to remove the images, containers, and volumes.  The easiest way is from the command line.  I started with removing the images using Docker desktop then moved to the command line.

Tyoe docker image ls to list all of the images then type docker image rm <and the ID of each image.  that will remove them.
Then type in docker volume ls to list the volume.  docker container rm <name> to remove them.

That should remove everything for you.

Paul

**Reply by Siggy101** [reply]
Thanks Paul.
I did some research and found essentially the same instructions.  
The only thing I am not certain of is which volumes to remove to really get the job done. 
I believe these are the relevant ones.  Can you confirm that's everything?  None of these containers are used for anything except TeslaMate

REPOSITORY                     TAG       IMAGE ID       CREATED        SIZE
postgres                       12        044aa8666500   5 days ago     314MB
eclipse-mosquitto              1.6       1feef7b48e2f   6 days ago     9.53MB
eclipse-mosquitto              latest    1feef7b48e2f   6 days ago     9.53MB
teslamate/teslamate            latest    bf11926b1075   7 weeks ago    44.2MB
teslamate/grafana              latest    3e6bcc2dd771   7 weeks ago    186MB

An uninstall guide would be valuable here @adriankumpf

**Reply by Adeian** [reply]
That will do it. Should get rid of all of it for you.

**Reply by jebinc** [reply]
guess the uninstall instructions never came.  Boy, is this software a mess on a raspberry pi

**Reply by natrlhy** [reply]
I run it flawlessly on a Pi. If following the instructions, it will run fine. Start over and try again step by step.

**Comment by ktsenvolt** [comment]
These would remove all the images, volumes, containers etc. So if you have other docker containers that you want to keep you will NOT want to run this.

#Stop the containers:
sudo docker-compose down 
#Delete all volumes:
docker volume rm $(docker volume ls -q) 
#Delete all containers:
docker rm -f $(docker ps -a -q) 
#Clean up any dangling branches
docker system prune 
#Remove all images from the local cache
docker rmi $(docker images -a -q)

**Comment by jebinc** [comment]
Thank youJohnOn Jan 23, 2023, at 6:13 AM, Ken ***@***.***> wrote:﻿
These would remove all the images, volumes, containers etc. So if you have other docker containers that you want to keep you will NOT want to run this.
#Stop the containers:
sudo docker-compose down
#Delete all volumes:
docker volume rm $(docker volume ls -q)
#Delete all containers:
docker rm -f $(docker ps -a -q)
#Clean up any dangling branches
docker system prune
#Remove all images from the local cache
docker rmi $(docker images -a -q)
